### PR TITLE
Add known issue for Java SDK

### PR DESCRIPTION
**Pull request by pauljewellmsft**

### CODE REVIEW

File: docs-ref-services/latest/storage.md [hunk 1]
```diff
@@ -355,4 +355,22 @@ The `BlobClient.upload(InputStream data)` method overwrites an existing blob by
...
+
+#### Issue description
+
+In the **azure-storage-file-share** package, the `ShareFileClient.downloadToFile()` method can write incorrect data to a file when more than 5 retries occur due to partial network responses while streaming data. The maximum number of retries is unintentionally higher (up to a maximum of 15), which can result in writing to incorrect positions in the file when the operation is retried more than 5 times. The data in Storage is still correct.
+
+#### Issue details
+
+| Client library | Versions impacted | Minimum safe version | Recommended action |
+| --- | --- | --- | --- |
+| Azure File Share | 12.15.0 to 12.25.0 | 12.26.0-beta.1 | [Update to latest version or minimum 12.26.0-beta.1](https://mvnrepository.com/artifact/com.azure/azure-storage-file-share) |
```
Comment (schoag-msft) [inline docs-ref-services/latest/storage.md:370]: we should call out 12.16.0 as the minimum safe version. We can't realistically expect customers to upgrade to a beta release

Comment (pauljewellmsft) [reply docs-ref-services/latest/storage.md]: Done. Updated to 12.26.0

Comment (seanmcc-msft) [reply docs-ref-services/latest/storage.md]: Let's change this to 12.25.1, the release today is a patch release.

File: docs-ref-services/latest/storage.md [hunk 2]
```diff
@@ -349,7 +349,25 @@ The `BlobClient.upload(InputStream data)` method overwrites an existing blob by
...
+
+Update client library versions according to the table above.
+
+[Back to list of known issues](#list-of-known-issues)
+
+### 12. Downloading with `ShareFileClient.downloadToFile()` can write incorrect data to a file
+
+#### Issue description
+
+In the **azure-storage-file-share** package, the `ShareFileClient.downloadToFile()` method can write incorrect data to a file when more than 5 retries occur due to partial network responses while streaming data. The maximum number of retries is unintentionally higher (up to a maximum of 15), which can result in writing to incorrect positions in the file when the operation is retried more than 5 times. The data in Storage is still correct.
```
Comment (seanmcc-msft) [inline docs-ref-services/latest/storage.md:364]: `The data in Storage is still correct.`

Maybe we should say "This issue does not affect data in the Storage service, only the data being written to the local filesystem."

Comment (pauljewellmsft) [reply docs-ref-services/latest/storage.md]: Updated. thx

File: docs-ref-services/latest/storage.md (hunk 1 above)
Comment (pauljewellmsft) [reply docs-ref-services/latest/storage.md]: Updated to 12.25.1 - thx for clarifying.

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**